test/functional/feature_reindex.py

```diff
@@ -68,12 +72,32 @@ def find_block(b, start):
             'LoadExternalBlockFile: Out of order block',
             'LoadExternalBlockFile: Processing out of order child',
         ]):
-            extra_args = [["-reindex"]]
-            self.start_nodes(extra_args)
+            self.start_node(0, ["-reindex"])
 
         # All blocks should be accepted and processed.
         assert_equal(self.nodes[0].getblockcount(), 12)
 
+    def reindex_readonly(self):
+        self.connect_nodes(0, 1)
+        # generate enough blocks to ensure that the -fastprune node fills up the
+        # first blk00000.dat file and starts another block file
+        self.generatetoaddress(self.nodes[1], 300, self.nodes[1].get_deterministic_priv_key().address)
```

Comment:
Verified:
```
$ ls -l /tmp/btest/node1/regtest/blocks/blk*.dat
-rw------- 1 larry larry 65400 Feb  3 16:58 /tmp/btest/node1/regtest/blocks/blk00000.dat
-rw------- 1 larry larry 16384 Feb  3 16:58 /tmp/btest/node1/regtest/blocks/blk00001.dat
```

Reply:
To ensure it, I think you could check if both files exist:
```diff
diff --git a/test/functional/feature_reindex.py b/test/functional/feature_reindex.py
index c68773e06..a3c9562f1 100755
--- a/test/functional/feature_reindex.py
+++ b/test/functional/feature_reindex.py
@@ -84,6 +84,9 @@ class ReindexTest(BitcoinTestFramework):
         self.generatetoaddress(self.nodes[1], 300, self.nodes[1].get_deterministic_priv_key().address)
         self.stop_node(1)
 
+        assert os.path.exists(self.nodes[1].chain_path / 'blocks' / 'blk00000.dat')
+        assert os.path.exists(self.nodes[1].chain_path / 'blocks' / 'blk00001.dat')
+
         # make the first block file read-only
         path = os.path.join(self.nodes[1].chain_path, 'blocks', 'blk00000.dat')
         os.chmod(path, 0o444)
```

Reply:
thanks, added.